src/main/java/reposense/wizard/WizardRunner.java

```diff
@@ -0,0 +1,54 @@
+package reposense.wizard;
+
+import static reposense.wizard.InputBuilder.translateCommandline;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.Scanner;
+
+import reposense.RepoSense;
+
+/**
+ * This class implements a wizard runner in the form
+ * of a discrete event simulator.
+ * It maintains a queue of prompts and run through
+ * the prompts until the queue is empty.
+ */
+public class WizardRunner {
+    private final Queue<Prompt> prompts;
+    private InputBuilder inputBuilder;
+
+    public WizardRunner(Wizard wizard) {
+        prompts = new ArrayDeque<>();
+        for (Prompt p : wizard.getInitialPrompts()) {
+            prompts.add(p);
+        }
+        inputBuilder = new InputBuilder();
+    }
+
+    /**
+     * Run the wizard until no more prompts is in the queue.
+     * If the wizard returns one or more prompts, add them
+     * to the queue, and repeat.
+     */
+    public void buildInput(Scanner sc) {
```

Comment:
Let's explore other options of getting input instead of passing in the scanner via buildInput, as it could be troublesome have to pass in the scanner whenever we use it.

Reply:
Agreed, but for the interest of time, I believe it's best to handle this in a future issue.